common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -355,6 +355,18 @@ public CalendarInterval subtract(CalendarInterval that) {
     return new CalendarInterval(months, microseconds);
   }
 
+  public CalendarInterval multiply(double num) {
+    int months = Math.toIntExact((long)(num * this.months));
```

Comment:
What happens when dividing an interval of 1 month by, say, 2? You'd end up with an interval of 0 time. I suppose the right answer is whatever PostgreSQL does.

Reply:
I will check what it does.

Reply:
PosgreSQL does this differently. I have re-implemented the operations.